card_generator/tasks/cards.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import tempfile
+
+from celery import shared_task
+from django.conf import settings
+from PyPDF2 import PdfMerger
+
+from card_generator.cards.client import QueueCardsClient
+from card_generator.cards.utils import convert_file_to_uri, data_uri_to_file
+
+logger = logging.getLogger()
+
+
+def get_pdfs(client: QueueCardsClient, batch_id: int):
+
+    raw_pdfs = client.get_id_queue_pdfs(batch_id)
+    return [item["id_pdf"] for item in raw_pdfs]
+
+
+def save_pdf_to_pds_spp(client: QueueCardsClient, batch_id: int, pdf_uri: str):
+    data = {"id_pdf": pdf_uri}
+    client.update_queue_batch_record(batch_id=batch_id, data=data)
+
+
+def merge_pdf(list_of_pdf: list, target_dir: str):
+    file_name = f"{target_dir}/result.pdf"
+    with PdfMerger() as merger:
+        for item in list_of_pdf:
+            merger.append(item)
+        merger.write(file_name)
+
+    return file_name
+
+
+@shared_task
+def merge_cards(batch_id: int):
+    client = QueueCardsClient(
+        server_root=settings.OPENSPP_SERVER_ROOT,
+        username=settings.OPENSPP_USERNAME,
+        password=settings.OPENSPP_API_TOKEN,
+        db_name=settings.OPENSPP_DB_NAME,
+    )
+
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
Are those temp dir cleanup after the data is not needed anymore?

Reply:
Yes @jeremi , the temp dir are cleaned up after exiting the context